scheduler/src/cook/mesos/util.clj

```diff
@@ -552,3 +552,26 @@
   [ch ch-size]
   (->> (repeatedly ch-size #(async/poll! ch))
        (remove nil?)))
+
+(defn xform-pipe
+  "Reads elements from the `in-chan` channel and supplies elements to the `out-chan` channel.
+   Maintains a state (initialized to `initial-state`) that is updated by applying the reducing function
+   `in-xform` to the current state and the incoming element `(in-xform state element)`.
+   `out-finalizer` accepts the current state and produces `[reset-state out-value]`.
+   When the out channel can receive an element, the `value` is put on the channel
+   and the state is set to the `reset-state` returned from the `out-finalizer`.
```

Comment:
You should add a note that out-finalizer may be called when the output is not consumed. (i.e. don't make it side effecty)